src/expr/macro/src/types.rs

```diff
@@ -17,61 +17,86 @@
 /// Expands a type wildcard string into a list of concrete types.
 pub fn expand_type_wildcard(ty: &str) -> Vec<&str> {
     match ty {
-        "*" => vec![
-            "boolean",
-            "int16",
-            "int32",
-            "int64",
-            "int256",
-            "float32",
-            "float64",
-            "decimal",
-            "serial",
-            "date",
-            "time",
-            "timestamp",
-            "timestamptz",
-            "interval",
-            "varchar",
-            "bytea",
-            "jsonb",
-            "struct",
-            "list",
-        ],
+        "*" => TYPE_MATRIX
+            .trim()
+            .lines()
+            .map(|l| l.split_whitespace().next().unwrap())
+            .collect(),
         "*int" => vec!["int16", "int32", "int64"],
         "*numeric" => vec!["decimal"],
         "*float" => vec!["float32", "float64"],
         _ => vec![ty],
     }
 }
 
-/// Maps a data type to its corresponding type name.
-pub fn to_data_type_name(ty: &str) -> Option<&str> {
-    Some(match ty {
-        "boolean" => "Boolean",
-        "int16" => "Int16",
-        "int32" => "Int32",
-        "int64" => "Int64",
-        "int256" => "Int256",
-        "float32" => "Float32",
-        "float64" => "Float64",
-        "decimal" => "Decimal",
-        "serial" => "Serial",
-        "date" => "Date",
-        "time" => "Time",
-        "timestamp" => "Timestamp",
-        "timestamptz" => "Timestamptz",
-        "interval" => "Interval",
-        "varchar" => "Varchar",
-        "bytea" => "Bytea",
-        "jsonb" => "Jsonb",
-        "struct" => "Struct",
-        "list" => "List",
-        _ => return None,
-    })
+//  name        data type   variant     array           owned           ref             primitive
+const TYPE_MATRIX: &str = "
```

Comment:
Can we have some words noticing developers to modify this after introducing new types?

Reply:
Added a note in the `risingwave_common::type` module.
https://github.com/risingwavelabs/risingwave/pull/9432/commits/d13f2bc3614f835f210d61d0d90ebc3c64946bee#diff-eaafd1f1b7bcfbb3f1f1a20e2064755bdd0867314889c0150ccb19e9f00dddf2